### PR TITLE
[FLOC-1635] Initial CLI for provisioning.

### PR DESCRIPTION
**Pull request by adamtheturtle**
https://clusterhq.atlassian.net/browse/FLOC-1635

### CODE REVIEW

**Comment by adamtheturtle** [comment]
I think that this should be continued with as it matches what has been signed off on by @robhaswell .

File: docs/indepth/installation.rst [hunk 1]
```diff
@@ -403,3 +403,35 @@ You have also ensured that the ``flocker-deploy`` command line tool is able to c
...
+Options are:
+* Cloud provider? Maybe
+   * For each cloud provider, credential options (e.g. rackspace secret key, region)
+* Number of nodes
+* Instance details - have defaults which work but user can decide instance size etc. - follow-up?
+
+For example:
+
+flocker-provision create --driver rackspace --rackspace-secret=secret --num-agent-nodes=3 --rackspace-region=us-west-1
+> Creating... 90%
```
Comment (adamtheturtle) [inline docs/indepth/installation.rst]: Will this progress report be possible? (If not) What helpful output could we give to show that something is going on?

**Comment by adamtheturtle** [comment]
We mentioned potentially having this tool output the YAML files necessary for the tutorial(s) but decided against this because the `flocker-deploy` tool may go away / change significantly soon. We should find out more detail on the plan and see how this tool could be useful for a user following the tutorial - it isn't fun changing all of the IP addresses in the YAML files and console commands.

File: docs/indepth/installation.rst [hunk 2]
```diff
@@ -403,3 +403,35 @@ You have also ensured that the ``flocker-deploy`` command line tool is able to c
 Next you may want to perform the steps in :doc:`the tutorial <./tutorial/moving-applications>` , to ensure that your nodes are correctly configured.
 Replace the IP addresses in the ``deployment.yaml`` files with the IP address of your own nodes.
 Keep in mind that the tutorial was designed with local virtual machines in mind, and results in an insecure environment.
+
+
+Experiment on a remote testing cluster
```
Comment (adamtheturtle) [inline docs/indepth/installation.rst]: Might be best to make it clear from the title how this differs from the above - one might spin up a Rackspace / CentOS 7 machine manually in order to experiment.

Also, the installation page is getting awfully long right now so maybe we should ask @alicesowerby about where this should go.

File: flocker/provision/script.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+from flocker.provision._install import (
+    configure_cluster,
+)
+
+
+class CreateOptions(Options):
+    # TODO longdesc and synopsis
+    # TODO look at docker machine for inspiration
+    optParameters = [
+        ['driver', 'd', 'rackspace', 'choose cloud provider'],
```
Comment (adamtheturtle) [inline flocker/provision/script.py]: Maybe not just cloud provider - this might spin up Vagrant VMs?

**Comment by tomprince** [comment]
This is on hold for now.

**Pull request closed without merging**